### PR TITLE
Update: Move template actions to the editor store.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Similar to https://github.com/WordPress/gutenberg/pull/60092.
This PR moves the template actions to the editor package.
It will allow actions like reverting a template to be available on the post editor (the post card panel at the top when editing the template will show the actions).

To move the "template actions" two store actions also needed to be included in the editor package.

## Testing

Open the templates section and verify actions and bulk actions still work as before.

### CODE REVIEW

**Comment by jorgefilipecosta** [comment]
cc: @mcsf as the approver and contributor of https://github.com/WordPress/gutenberg/pull/60092 a related PR.

File: packages/edit-site/src/store/actions.js [hunk 1]
```diff
@@ -133,9 +128,13 @@ export const addTemplate =
  *
  * @param {Object} template The template object.
  */
-export const removeTemplate = ( template ) => {
-	return removeTemplates( [ template ] );
-};
+export const removeTemplate =
+	( template ) =>
+	async ( { registry } ) => {
+		await unlock( registry.dispatch( editorStore ) ).removeTemplates( [
```
Comment (mcsf) [inline packages/edit-site/src/store/actions.js]: (minor question) Is the `async/await` structure necessary here? We're just delegating to a private action, so I'd've expected something like `removeTemplate = tpl => unlock( ... ).removeTemplates( [ tpl ] )`. Or is it to signal something to the reader?

File: packages/editor/package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
 		"@wordpress/compose": "file:../compose",
 		"@wordpress/core-data": "file:../core-data",
 		"@wordpress/data": "file:../data",
+		"@wordpress/data-controls": "file:../data-controls",
```
Comment (mcsf) [inline packages/editor/package.json]: What is this for?

File: packages/editor/src/components/post-actions/actions.js [hunk 3]
```diff
@@ -491,6 +499,273 @@ export const renamePostAction = {
 	},
 };
 
+export const resetTemplateAction = {
```
Comment (mcsf) [inline packages/editor/src/components/post-actions/actions.js]: (I don't remember if I have already asked this question) Why are we exporting these actions? Unless there is a present need, let's hide them and only expose `usePostActions`, no?

**Review by mcsf — APPROVED** [review]
Left some notes and a recommendation, but other than that LGTM!

File: packages/edit-site/src/store/actions.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/store/actions.js]: Yes, we can just return and I updated the code. But it is a fact that async await provides a better signal to the reader that the action is async, but we don't do it in other parts of the code so we can just return here as well.

File: packages/editor/package.json (hunk 2 above)
Comment (jorgefilipecosta) [reply packages/editor/package.json]: The IDE automatically added a wrong import the line was removed.

File: packages/editor/src/components/post-actions/actions.js (hunk 3 above)
Comment (jorgefilipecosta) [reply packages/editor/src/components/post-actions/actions.js]: Yes, I missed this on the other PR, the exports were removed.

**Pull request merged**